api/src/services/AuthService.ts

```diff
@@ -76,17 +79,19 @@ export class AuthService {
         }
     }
 
-    registerWithProvider = async (profile) => {
+    registerWithProvider = async profile => {
         const exitingUser = await this.userRepository.findOne({
             email: profile.email,
         })
         if (exitingUser) {
-            throw [{ error: 'email aleardy exists' }]
+            return {
+                token: this.encryptionService.sign(classToPlain(exitingUser)),
```

Comment:
accept the sign method in EncryptionService to accept either JSON or an User model and do the classToPlain in there if existingUser is an instance of User.